Archives/ClmFile.cpp

```diff
@@ -405,24 +390,57 @@ namespace Archives
 		return strippedExtensions;
 	}
 
+	WaveFormatEx ClmFile::CreateDefaultWaveFormat() {
+		WaveFormatEx waveFormat;
+
+		waveFormat.wFormatTag = 1; // WAVE_FORMAT_PCM
+		waveFormat.nChannels = 1; // mono
+		waveFormat.nSamplesPerSec = 22050; // 22.05KHz
+		waveFormat.nAvgBytesPerSec = 44100; // nSamplesPerSec *nBlockAlign
+		waveFormat.nBlockAlign = 2; // 2 bytes/sample = nChannels * wBitsPerSample / 8
+		waveFormat.wBitsPerSample = 16;
+		waveFormat.cbSize = 0;
+
+		return waveFormat;
+	}
+
 
 	// A null terminator (\0) is automatically assigned to the end of the string when placing it within the std::array
 	const std::array<char, 32> standardFileVersion { "OP2 Clump File Version 1.0\x01A\0\0\0\0" };
 	const std::array<char, 6> standardUnknown { 0, 0, 0, 0, 1, 0 };
 
 	ClmFile::ClmHeader::ClmHeader(WaveFormatEx waveFormat, uint32_t packedFilesCount) {
-		std::memcpy(fileVersion.data(), standardFileVersion.data(), fileVersion.size());
-		std::memcpy(unknown.data(), standardUnknown.data(), unknown.size());
+		fileVersion = standardFileVersion;
+		unknown = standardUnknown;
 
 		this->waveFormat = waveFormat;
 		this->packedFilesCount = packedFilesCount;
 	}
 
 	bool ClmFile::ClmHeader::CheckFileVersion() const {
-		return std::memcmp(fileVersion.data(), standardFileVersion.data(), fileVersion.size()) == 0;
+		return fileVersion == standardFileVersion;
 	};
 
 	bool ClmFile::ClmHeader::CheckUnknown() const {
-		return std::memcmp(unknown.data(), standardUnknown.data(), unknown.size()) == 0;
+		return unknown == standardUnknown;
+	}
+
+	void ClmFile::ClmHeader::VerifyFileVersion() const {
+		if (!CheckFileVersion()) {
+			throw std::runtime_error("CLM file version is incorrect.");
```

Comment:
Might make sense to include the CLM filename here, if it's available.

Reply:
I left the CLM's filename outside of the a ClmHeader for 2 reasons. 

1. It would add an extra data field that doesn't exist in an actual CLM file and complicate reading/writing a bit. 
2. When packing a new CLM file, the filename may not exist when the ClmHeader is created (well, this is only partially true since you probably want to specifiy the planned CLM filename before constructing it in memory.)

I'd like to leave the CLM's filename outside of the clmHeader struct.